internal/httputil/pagination.go

```diff
@@ -0,0 +1,58 @@
+package httputil
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+const maxPageSize = 100
+
+type PaginationData struct {
+	PageNumber int
+	PageSize   int
+	SortBy     map[string]string
+	FilterBy   map[string]string
+}
+
+func ParsePaginationQuery(r *http.Request) (*PaginationData, error) {
+	qs := r.URL.Query()
+
+	pagination := &PaginationData{
```

Comment:
It would be better: pagination := &PaginationData{}, because we will never return assigned values in this struct.